corehq/apps/hqadmin/management/commands/clone_domain.py

```diff
@@ -175,9 +176,13 @@ def copy_locations(self, types_only=False):
             location_types_map[old_id] = new_id
 
         if not types_only:
-            # MPTT sorts this queryset so we can just save in the same order
+            # use get_descendants, which sorts locations hierarchically,
+            # so we can save in the same order
```

Comment:
How difficult would it be to always use this order?  Does it introduce overhead on the query?  I wonder if we have other random places which implicitly depend on getting results in tree order.

Reply:
I'm not sure I understand the question. It sounds like you're asking if we could have the default ordering be tree order (as was previously done in `LocationManager.get_queryset()`). To do that we'd have to make every query use a recursive CTE. Queries that are known to access only one level of the tree could avoid the CTE, but detecting that automatically sounds hard. This would add a lot of overhead which would affect our postgres query execution capacity, and it would be realized in many places where ordering is not important.

Reply:
Yeah, that's kinda what I figured.  Seems like an acceptable risk to take, given the difficulty of always using tree order.  Would it be reasonably simple and performant to order by depth, or is would that also require recursive queries to compute?  By "depth", I mean all states, then all counties, then all cities, so not tree order, but it'd meet the needs of this use-case.  Figured I'd mention in case that sounds more performant or easier than a proper tree order like this.